packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,208 @@
+/// of the [ValueKey] class. This key is what will be returned in
+/// the inputted callback.
+///
+/// See also:
+///
+/// * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
+class SegmentedControl extends StatelessWidget {
+  /// Creates an iOS-style segmented control bar.
+  SegmentedControl(
+      {Key key,
+      @required this.children,
+      @required this.onValueChanged,
+      this.value})
+      : assert(children != null),
+        assert(children.length >= 2),
+        assert(onValueChanged != null),
+        super(key: key);
+
+  /// The list of widgets of that will be used to populate the tabs of the bar.
+  final List<Widget> children;
+
+  /// The Key value of the widget that should be initially selected.
+  ///
+  /// This Key should be the Key of one of the widgets within [children].
+  final Key value;
+
+  /// The callback called when an option is pressed.
+  ///
+  /// This callback should set the current Key value equal to the returned Key.
+  final ValueChanged<Key> onValueChanged;
+
+  List<Widget> _buildSegmentedControlButtons() {
+    List<Widget> buttons = new List<Widget>();
+
+    for (int i = 0; i < children.length; i += 1) {
+      buttons.add(new Expanded(
+          child: new _SegmentedControlButton(
+              children.elementAt(i),
+              children.elementAt(i).key,
+              onValueChanged,
+              value,
+              false,
+              i == 0,
+              i == children.length - 1)));
+
+      if (i != children.length - 1) {
+        buttons.add(new _SegmentedControlButton(
+            new Container(
+                decoration:
+                    const BoxDecoration(color: CupertinoColors.activeBlue),
+                constraints:
+                    const BoxConstraints(maxWidth: 2.0, minWidth: 2.0)),
+            children.elementAt(i).key,
+            onValueChanged,
+            value,
+            true,
+            false,
+            false));
+      }
+    }
+    return buttons;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: _kHorizontalItemPadding,
+      child: new Align(
+        child: new Container(
+          decoration: const ShapeDecoration(
+            color: CupertinoColors.white,
+            shape: const RoundedRectangleBorder(
+              side: BorderSide(
+                  color: CupertinoColors.activeBlue,
+                  width: 2.0,
+                  style: BorderStyle.solid),
+              borderRadius: BorderRadius.all(_kDefaultCornerRadius),
+            ),
+          ),
+          child: new Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: _buildSegmentedControlButtons()),
+        ),
+      ),
+    );
+  }
+}
+
+class _SegmentedControlButton extends StatefulWidget {
+  _SegmentedControlButton(this.selectState, this.widgetKey, this.onValueChanged,
```

Comment:
These arguments should all be named parameters instead of positional parameters.  You should add `@required` in front of each one, and assert that they are all non-null (if that is the contract).